expr/sort.go

```diff
@@ -31,8 +31,15 @@ func (b byPartBase) Swap(i, j int) {
 }
 
 func getPart(metric *types.MetricData, part int) string {
+	// metric name should not contain braces
+	// we can use normal split here
 	parts := strings.Split(metric.Name, ".")
-	return parts[part]
+	// returning proper part if in range
+	if part < len(parts) {
+		return parts[part]
+	}
+	// TODO: that should never happen, maybe we should log it
```

Comment:
If it's clear that this execution path is wrong and shouldn't happen, it makes sense to return an error. That would be safer: we will make the unexpected behavior evident, and the error will clearly indicate what is wrong.

Reply:
Good idea, but I'm not sure how to do so - as you can see `getPart` is used in `sort.Stable(AlphabeticallyByPart(metrics, i))` and `sort.Stable` do not generate or propagate errors. I will mention that in TODO, though.